pkg/cmd/admin/router/router.go

```diff
@@ -552,24 +552,46 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out, errout io.Write
 	var clusterIP string
 
 	output := cfg.Action.ShouldPrint()
-	generate := output
+	generateSvc := output
 	service, err := kClient.Services(namespace).Get(name)
 	if err != nil {
-		if !generate {
+		if !generateSvc {
 			if !errors.IsNotFound(err) {
 				return fmt.Errorf("can't check for existing router %q: %v", name, err)
 			}
 			if !output && cfg.Action.DryRun {
 				return fmt.Errorf("Router %q service does not exist", name)
 			}
-			generate = true
+			generateSvc = true
 		}
 	} else {
 		clusterIP = service.Spec.ClusterIP
+		fmt.Fprintf(out, "Router %q service exists\n", name)
 	}
 
-	if !generate {
-		fmt.Fprintf(out, "Router %q service exists\n", name)
+	certName := fmt.Sprintf("%s-certs", cfg.Name)
+	generateSecret := output
+	_, err = kClient.Secrets(namespace).Get(certName)
```

Comment:
all of this should move down after generateSecretsConfig is called and operate on the slice of secrets returned